### PR TITLE
Best attempt to get HotelRoom routes

### PR DESCRIPTION
**Pull request by OCDAmmo3**

### CODE REVIEW

**Comment by OCDAmmo3** [comment]
Currently broken - GetHotelRooms in HotelsController. Does not resolve properly. Cannot figure out what to put in as the return value, or what to be returning. Will resolve when that is known.

File: Hotels/Controllers/HotelsController.cs [hunk 1]
```diff
@@ -88,5 +88,29 @@ public async Task<ActionResult<Hotel>> DeleteHotel(long id)
 
             return hotel;
         }
+
+        // GET: api/Hotels/5/Rooms
+        [HttpGet("{id}/Rooms")]
+        public async Task<ActionResult<List<HotelRoom>>> GetHotelRooms(long hotelId)
```
Comment (dahlbyk) [inline Hotels/Controllers/HotelsController.cs]:
```suggestion
        [HttpGet("{hotelId}/Rooms")]
        public async Task<ActionResult<List<HotelRoom>>> GetHotelRooms(long hotelId)
```
or
```suggestion
        [HttpGet("{id}/Rooms")]
        public async Task<ActionResult<List<HotelRoom>>> GetHotelRooms(long id)
```

File: Hotels/Controllers/HotelsController.cs [hunk 2]
```diff
@@ -88,5 +88,29 @@ public async Task<ActionResult<Hotel>> DeleteHotel(long id)
...
+        {
+            var hotel = await repository.GetHotelRoomsById(hotelId);
+            return hotel.HotelRooms;
+        }
+
+        // POST: api/Hotels/5/Rooms/12
+        [HttpPost("{hotelId}/Rooms/{roomId}")]
+        public async Task<ActionResult> AddRoom(long hotelId, long roomId)
+        {
+            await repository.AddRoomAsync(hotelId, roomId);
```
Comment (dahlbyk) [inline Hotels/Controllers/HotelsController.cs]: Too easy to confuse `roomId` with `roomNumber`
```suggestion
        [HttpPost("{hotelId}/Rooms")]
        public async Task<ActionResult> AddRoom(long hotelId, CreateHotelRoom hotelRoom)
        {
            await repository.AddRoomAsync(hotelId, hotelRoom);
```

`CreateHotelRoom` would look something like this:

https://github.com/DeltaVCode/cr-dotnet-401d2/blob/b88bf612a03e135f94dd7e08f687e5ec093fbbdd/Class14/Demo/Web/Models/CreateGrade.cs#L5-L12

File: Hotels/Controllers/HotelsController.cs [hunk 3]
```diff
@@ -88,5 +88,29 @@ public async Task<ActionResult<Hotel>> DeleteHotel(long id)
...
+            return CreatedAtAction(nameof(AddRoom), new { hotelId, roomId }, null);
+        }
+
+        // DELETE: api/Hotels/5/Rooms/12
+        [HttpDelete("{hotelId}/Rooms/{roomId}")]
+        public async Task<ActionResult> DeleteRoom(long hotelId, long roomId)
```
Comment (dahlbyk) [inline Hotels/Controllers/HotelsController.cs]: This route should receive `roomNumber` not `roomId`
```suggestion
        [HttpDelete("{hotelId}/Rooms/{roomNumber}")]
        public async Task<ActionResult> DeleteRoom(long hotelId, long roomNumber)
```

**Review by dahlbyk — COMMENTED** [review]
Class Review

**Pull request merged**